test/test_parabolic_2d.jl

```diff
@@ -52,20 +52,22 @@ isdir(outdir) && rm(outdir, recursive = true)
 
     t = 0.0
     # pass in `boundary_condition_periodic` to skip boundary flux/integral evaluation
-    Trixi.calc_gradient!(gradients, ode.u0, t, mesh, equations_parabolic,
+    Trixi.calc_gradient!(gradients, Base.parent(ode.u0), t, mesh, equations_parabolic,
```

Comment:
Would it make sense to define a new method, which dispatches on `VectorOfArray` and forwards this to the currently existing `calc_gradient!` by passing `Base.parent(u_ode)` and keeping the remaining arguments unchanged?  This way we wouldn't need to change anything here.

Reply:
Since this function is just intended to be called internally after calling `wrap_array`, I would avoid adding another function specialization (the dispatch graph is complicated enough as-is already). 

How about instead I clean the test up slightly so `u0 = Base.parent(ode.u0)` is only unpacked once in these tests?

Reply:
Ok, sounds good!